packages/sync-workflows/activities/log_sync_finish.ts

```
@@ -32,14 +32,21 @@ export function createLogSyncFinish({
     numRecordsSynced: number | null;
   }) {
     await syncHistoryService.logFinish({ historyId, status, errorMessage, numRecordsSynced });
+
+    const connection = await connectionService.getSafeById(connectionId);
+    const application = await applicationService.getById(connection.applicationId);
+
     if (status === 'FAILURE') {
       const error = new Error(errorMessage);
       error.stack = errorStack;
-      logger.error(error, `Sync failed for syncId ${syncId} and connectionId ${connectionId}`);
+
+      if (application.environment === 'development') {
```

Do we want to swallow these errors for self-hosted as well? Or just when `IS_CLOUD` is set?